### PR TITLE
Add basic functionality to login page

### PR DESCRIPTION
**Pull request by Linzeur**
- Create two function, one return the object general to use in all application (`newAppObject`), and the another is to save this object in the storage (`createUser`)
- Add the load event where it validates if the storage exists, on the contrary, call a new function named `assignEvents` where it adds the events to the unique button which exists in the login page, and the sockets events to future use as issue #27.

### CODE REVIEW

File: public/login.html [hunk 1]
```diff
@@ -41,13 +41,14 @@
...
           class="input-username"
           type="text"
-          name="username"
+          id="txtUsername"
           placeholder="How do you want to be called?"
           title="You can change this later."
         />
         <br />
-        <input class="button-login" type="submit" value="Continue"></input>
+        <input class="button-login" type="submit" id="btnContinue" value="Continue"></input>
```
Comment (Gatts1) [inline public/login.html]: Why did you add an id and you don't just use a class?

**Comment by paulotijero** [comment]
@Linzeur Please, remove your package-lock.json

File: public/assets/css/login.css [hunk 2]
```diff
@@ -84,6 +84,7 @@ body {
 }
 .data-info span {
   color: #b9bbbe;
+<<<<<<< HEAD
```
Comment (paulotijero) [inline public/assets/css/login.css]: Be careful when doing rebase, remove this line :eyes:

File: public/assets/css/login.css [hunk 3]
```diff
@@ -120,3 +121,8 @@ body {
   background-color: #7289da;
   cursor: pointer;
 }
+=======
```
Comment (paulotijero) [inline public/assets/css/login.css]: This line also :eyes:

File: public/assets/css/login.css [hunk 4]
```diff
@@ -120,3 +121,8 @@ body {
...
+  font-size: 16px;
+  white-space: nowrap;
+}
+>>>>>>> Add container login
```
Comment (paulotijero) [inline public/assets/css/login.css]: This line also :eyes:

File: public/assets/css/login.css (hunk 2 above)
Comment (Linzeur) [reply public/assets/css/login.css]: Oh, you're right! I will fix that, thanks

File: public/assets/js/signIn.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+const keyStorage = "app-data",
+  pageToRedirect = "index.html";
```
Comment (paulotijero) [inline public/assets/js/signIn.js]: How do function this? It's same that this?
```js
const keyStorage = "app-data";
const pageToRedirect = "index.html";
```

File: public/login.html (hunk 1 above)
Comment (Linzeur) [reply public/login.html]: Because I only need one element which I want to add an event, if I look for this by class, it return by default a list, but it would have one element, and I would have that access to it through index. With the `getElementById` is directly

File: public/assets/js/signIn.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    currentuser: {
+      id: Date.now(),
+      username: newUsername,
+      creationDate: new Date(),
+      state: "active"
+    },
+    users: [],
+    channels: [
+      {
+        id: 1000000000000,
```
Comment (paulotijero) [inline public/assets/js/signIn.js]: This `id: 1000000000000,` Only use for the tests or which is the function?

File: public/assets/js/signIn.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    console.log("Connection closed");
+  });
+  socket.addEventListener("message", event => {
+    console.log(JSON.parse(event.data));
+  });
+
+  let btnContinue = document.getElementById("btnContinue");
+  btnContinue.addEventListener("click", function() {
+    createUser();
+    window.location.href = pageToRedirect;
```
Comment (diegotc86) [inline public/assets/js/signIn.js]: Hi Brayan. Why did you create a new call back as second parameter of the `addEventListener` method. It because you not only call one function and instead you do more things?
For example: `btnContinue.addEventListener("click", handlerFunction)`

File: public/assets/js/signIn.js (hunk 5 above)
Comment (Linzeur) [reply public/assets/js/signIn.js]: Yep, only that I saw unnecessary separate to write twice the keyword const, also all variables are of the same data type, I would prefer to group all variable of the same type with the same keyword to declare.

File: public/assets/js/signIn.js (hunk 6 above)
Comment (Linzeur) [reply public/assets/js/signIn.js]: I put that because the general channel must be same for everyone, if I would put that it generates by something, how an user can join to this in different  times for first time, it could generate a issue because the id of general channel would be different

File: public/login.html (hunk 1 above)
Comment (paulotijero) [reply public/login.html]: You right, but could be better add a name for the input because this improves a11y

File: public/assets/js/signIn.js (hunk 7 above)
Comment (Linzeur) [reply public/assets/js/signIn.js]: Right, it had been more easy to put the line 52 within `createUser` function, but when we need to do test, it would be a little problematic to do that. Also, it only does two things and create a function to only this, I think is unnecessary
